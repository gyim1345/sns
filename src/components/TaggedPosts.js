import React, { useState, useEffect } from "react";
import { taggedPostsAPI } from "../apis/post";
import { Link } from "react-router-dom";
import PostImagesOnly from "./PostImagesOnly";
import { css } from "@emotion/core";

function TaggedPosts({ user }) {
  const [posts, setPosts] = useState([]);

  const getTaggedPost = async () => {
    try {
      const response = await taggedPostsAPI(user);
      setPosts(response);
    } catch (e) {
      console.log(e);
    }
  };

  useEffect(() => {
    getTaggedPost();
  }, []);

  return (
    <div css={[tagPostImagesCss]}>
      {posts.map(post => (
        <Link to={`/${post.userName}/posting/${post.id}`} key={post.id}>
          <PostImagesOnly imageUrl={post.imageUrl} />
        </Link>
      ))}
    </div>
  );
}

const tagPostImagesCss = css`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  flex-basis: 33%;
  justify-content: center;
`;

export default TaggedPosts;
