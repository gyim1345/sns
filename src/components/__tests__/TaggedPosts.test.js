import React from 'react';
import { mount } from 'enzyme';
import TaggedPosts from '../TaggedPosts';
import { currentUser } from './testjs';

describe('<TaggedPosts />', () => {
  it('스냅샷 비교', () => {
    const wrapper = mount(<TaggedPosts user={currentUser} />);
    expect(wrapper).toMatchSnapshot();
  });
});
