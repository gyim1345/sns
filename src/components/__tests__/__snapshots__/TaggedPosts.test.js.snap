// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaggedPosts /> 스냅샷 비교 1`] = `
<TaggedPosts
  user="gibong@gmail.com"
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="TaggedPosts"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlRhZ2dlZFBvc3RzLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQXdDNEIiLCJmaWxlIjoiVGFnZ2VkUG9zdHMuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QsIHsgdXNlU3RhdGUsIHVzZUVmZmVjdCB9IGZyb20gJ3JlYWN0JztcbmltcG9ydCB7IExpbmsgfSBmcm9tICdyZWFjdC1yb3V0ZXItZG9tJztcbmltcG9ydCBTd2FsIGZyb20gJ3N3ZWV0YWxlcnQyJztcblxuaW1wb3J0IHsgY3NzIH0gZnJvbSAnQGVtb3Rpb24vY29yZSc7XG5cbmltcG9ydCB7IHRhZ2dlZFBvc3RzQVBJIH0gZnJvbSAnLi4vYXBpcy9wb3N0JztcbmltcG9ydCBQb3N0SW1hZ2VzT25seSBmcm9tICcuL1Bvc3RJbWFnZXNPbmx5JztcblxuZnVuY3Rpb24gVGFnZ2VkUG9zdHMoeyB1c2VyIH0pIHtcbiAgY29uc3QgW3Bvc3RzLCBzZXRQb3N0c10gPSB1c2VTdGF0ZShbXSk7XG5cbiAgY29uc3QgZ2V0VGFnZ2VkUG9zdCA9IGFzeW5jICgpID0+IHtcbiAgICB0cnkge1xuICAgICAgY29uc3QgcmVzcG9uc2UgPSBhd2FpdCB0YWdnZWRQb3N0c0FQSSh1c2VyKTtcbiAgICAgIHNldFBvc3RzKHJlc3BvbnNlKTtcbiAgICB9IGNhdGNoIChlKSB7XG4gICAgICBTd2FsLmZpcmUoe1xuICAgICAgICBpY29uOiAnZXJyb3InLFxuICAgICAgICB0aXRsZTogJ09vcHMuLi4nLFxuICAgICAgICB0ZXh0OiAnSW50ZXJuYWwgRXJyb3InXG4gICAgICB9KTtcbiAgICB9XG4gIH07XG5cbiAgdXNlRWZmZWN0KCgpID0+IHtcbiAgICBnZXRUYWdnZWRQb3N0KCk7XG4gIH0sIFtdKTtcblxuICByZXR1cm4gKFxuICAgIDxkaXYgY3NzPXtbdGFnUG9zdEltYWdlc0Nzc119PlxuICAgICAge3Bvc3RzLm1hcChwb3N0ID0+IChcbiAgICAgICAgPExpbmsgdG89e2AvcG9zdGluZy8ke3Bvc3QuaWR9YH0ga2V5PXtwb3N0LmlkfT5cbiAgICAgICAgICA8UG9zdEltYWdlc09ubHkgaW1hZ2VVcmw9e3Bvc3QuaW1hZ2VVcmx9IC8+XG4gICAgICAgIDwvTGluaz5cbiAgICAgICkpfVxuICAgIDwvZGl2PlxuICApO1xufVxuXG5jb25zdCB0YWdQb3N0SW1hZ2VzQ3NzID0gY3NzYFxuICBkaXNwbGF5OiBmbGV4O1xuICBmbGV4LWRpcmVjdGlvbjogcm93O1xuICBmbGV4LXdyYXA6IHdyYXA7XG4gIGZsZXgtYmFzaXM6IDMzJTtcbiAganVzdGlmeS1jb250ZW50OiBjZW50ZXI7XG5gO1xuXG5leHBvcnQgZGVmYXVsdCBUYWdnZWRQb3N0cztcbiJdfQ== */",
        "name": "1qzsve-tagPostImagesCss-TaggedPosts",
        "next": undefined,
        "styles": "display:flex;flex-direction:row;flex-wrap:wrap;flex-basis:33%;justify-content:center;;label:tagPostImagesCss;;;label:TaggedPosts;/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlRhZ2dlZFBvc3RzLmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQThCUyIsImZpbGUiOiJUYWdnZWRQb3N0cy5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBSZWFjdCwgeyB1c2VTdGF0ZSwgdXNlRWZmZWN0IH0gZnJvbSAncmVhY3QnO1xuaW1wb3J0IHsgTGluayB9IGZyb20gJ3JlYWN0LXJvdXRlci1kb20nO1xuaW1wb3J0IFN3YWwgZnJvbSAnc3dlZXRhbGVydDInO1xuXG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcblxuaW1wb3J0IHsgdGFnZ2VkUG9zdHNBUEkgfSBmcm9tICcuLi9hcGlzL3Bvc3QnO1xuaW1wb3J0IFBvc3RJbWFnZXNPbmx5IGZyb20gJy4vUG9zdEltYWdlc09ubHknO1xuXG5mdW5jdGlvbiBUYWdnZWRQb3N0cyh7IHVzZXIgfSkge1xuICBjb25zdCBbcG9zdHMsIHNldFBvc3RzXSA9IHVzZVN0YXRlKFtdKTtcblxuICBjb25zdCBnZXRUYWdnZWRQb3N0ID0gYXN5bmMgKCkgPT4ge1xuICAgIHRyeSB7XG4gICAgICBjb25zdCByZXNwb25zZSA9IGF3YWl0IHRhZ2dlZFBvc3RzQVBJKHVzZXIpO1xuICAgICAgc2V0UG9zdHMocmVzcG9uc2UpO1xuICAgIH0gY2F0Y2ggKGUpIHtcbiAgICAgIFN3YWwuZmlyZSh7XG4gICAgICAgIGljb246ICdlcnJvcicsXG4gICAgICAgIHRpdGxlOiAnT29wcy4uLicsXG4gICAgICAgIHRleHQ6ICdJbnRlcm5hbCBFcnJvcidcbiAgICAgIH0pO1xuICAgIH1cbiAgfTtcblxuICB1c2VFZmZlY3QoKCkgPT4ge1xuICAgIGdldFRhZ2dlZFBvc3QoKTtcbiAgfSwgW10pO1xuXG4gIHJldHVybiAoXG4gICAgPGRpdiBjc3M9e1t0YWdQb3N0SW1hZ2VzQ3NzXX0+XG4gICAgICB7cG9zdHMubWFwKHBvc3QgPT4gKFxuICAgICAgICA8TGluayB0bz17YC9wb3N0aW5nLyR7cG9zdC5pZH1gfSBrZXk9e3Bvc3QuaWR9PlxuICAgICAgICAgIDxQb3N0SW1hZ2VzT25seSBpbWFnZVVybD17cG9zdC5pbWFnZVVybH0gLz5cbiAgICAgICAgPC9MaW5rPlxuICAgICAgKSl9XG4gICAgPC9kaXY+XG4gICk7XG59XG5cbmNvbnN0IHRhZ1Bvc3RJbWFnZXNDc3MgPSBjc3NgXG4gIGRpc3BsYXk6IGZsZXg7XG4gIGZsZXgtZGlyZWN0aW9uOiByb3c7XG4gIGZsZXgtd3JhcDogd3JhcDtcbiAgZmxleC1iYXNpczogMzMlO1xuICBqdXN0aWZ5LWNvbnRlbnQ6IGNlbnRlcjtcbmA7XG5cbmV4cG9ydCBkZWZhdWx0IFRhZ2dlZFBvc3RzO1xuIl19 */",
        "toString": [Function],
      }
    }
  >
    <div
      className="css-1qzsve-tagPostImagesCss-TaggedPosts"
    />
  </EmotionCssPropInternal>
</TaggedPosts>
`;
